UT4MasterServer.Xmpp/XmppServer.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Net;
+using System.Net.Sockets;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using UT4MasterServer.Xmpp.Stanzas;
+
+namespace UT4MasterServer.Xmpp;
+
+public class XmppServer
+{
+	private TcpListener listener;
+	private List<XmppConnection> connections;
+
+	private DateTimeOffset lastAcceptTime = default;
+	private readonly TimeSpan onlyAcceptEvery = TimeSpan.FromSeconds(1);
+
+	public string Domain { get; private set; }
+	public JID AdminID { get; private set; }
+	public X509Certificate2 Certificate { get; private set; }
+
+	public XmppServer(string domain, X509Certificate2 certificate)
+	{
+		Domain = domain;
+		AdminID = new JID("xmpp-admin", Domain);
```

Comment:
This value should probably go to appsettings.

Reply:
game has this hardcoded as admin username, so not sure if that's really required